time.c

```diff
@@ -4826,6 +4828,102 @@ time_to_a(VALUE time)
                     time_zone(time));
 }
 
+#define set_key(key_id, val) \
+        rb_hash_aset(h, ID2SYM(key_id), val);
+
+#define set_key_if_matches(sym_id, val) \
+        if(ID2SYM(sym_id) == key) rb_hash_aset(h, key, val);
```

Comment:
I'd prefer to not use macros here, as I think it makes the resulting code more difficult to understand.

Reply:
I'll change it if you insist, but honestly, I believe that repeating 11 (eleven) times the construct 
```c
if(ID2SYM(id_year) == key) {
  rb_hash_aset(h, key, tobj->vtm.year);
}
```
...seems to me the way to make it harder to understand, not easier. Currently, it is 100% exposed that we just do the very same thing 11 times.
But with all the `if`s (properly formatted like C codebase prefers) it would be 33 lines which will require some head-scratching to understand whether they are homogenous or not, and whether all of them are just setting the field by condition or there are some more details to follow.

Reply:
Certainly the `set_key` macro is not helpful over calling `rb_hash_aset` directly, especially once you switch from static IDs to static symbols.

In terms of `set_key_if_matches`, that is a grayer area.  However, you don't need 33 lines. you can keep the same 11 lines.  Ruby uses `if (cond) do_this;` (single line without braces) all the time (simple grep shows over 4000 uses just in the `*.c` in the root directory).

If you think it must be kept, please move the macro definition to the line before first use, and the undef to the line after last use.  That at least reduces possible confusion.